core/src/test/java/io/questdb/test/std/str/DirectByteCharSinkTest.java

```diff
@@ -24,15 +24,126 @@
 
 package io.questdb.test.std.str;
 
+import io.questdb.std.MemoryTag;
 import io.questdb.std.Unsafe;
+import io.questdb.std.bytes.NativeByteSink;
+import io.questdb.std.str.ByteSequence;
 import io.questdb.std.str.DirectByteCharSink;
 import io.questdb.test.tools.TestUtils;
 import org.junit.Assert;
 import org.junit.Test;
 
 import java.nio.charset.StandardCharsets;
+import java.util.function.Supplier;
 
 public class DirectByteCharSinkTest {
+    @Test
+    public void testBorrowDirectByteSink() throws Exception {
+        try (DirectByteCharSink sink = new DirectByteCharSink(16)) {
+            final long ptr = sink.ptr();
+            Assert.assertNotEquals(0, ptr);
+            Assert.assertEquals(32, sink.getCapacity());
+            sink.put((byte) 'a');
+            sink.put((byte) 'b');
+            sink.put((byte) 'c');
+            Assert.assertEquals(3, sink.length());
+            Assert.assertEquals((byte) 'a', sink.byteAt(0));
+            Assert.assertEquals((byte) 'b', sink.byteAt(1));
+            Assert.assertEquals((byte) 'c', sink.byteAt(2));
+
+            try (NativeByteSink directSink = sink.borrowDirectByteSink()) {
+                final long impl = directSink.ptr();
+                Assert.assertNotEquals(ptr, impl);
+                final long implPtr = Unsafe.getUnsafe().getLong(impl);
+                Unsafe.getUnsafe().putByte(implPtr, (byte) 'd');
+                Unsafe.getUnsafe().putLong(impl, implPtr + 1);
+                Assert.assertEquals(4, sink.length());
+                Assert.assertEquals(32, sink.getCapacity());
+                final long newImplPtr = NativeByteSink.book(impl, 400);
```

Comment:
This looks like a guts-out API. `sink.borrowDirectByteSink()` shouldn't return a class with a bunch of native methods like `create()` or `destroy()`. Instead, it should return something with `ptr()`, `pos()` and `size()` methods only, so that it's not easy for the caller to do something wrong.

Reply:
All the class does is expose a pointer for passing onto native code:
* A `ptr()` method that yields the `questdb_byte_sink` in C.
* A `close()` method so the owner object can perform its accounting.

I'll move them to as `@TestOnly` methods and into the `DirectByteSink` class. Hopefully that should resolve the confusion.

Reply:
Well still need the `ptr()` ... new interface is effectively that:

```
/**
 * A try-with-resource accessor to the `questdb_byte_sink_t` structure.
 * <p>
 * Note that the close method is simply meant to allow the owner object to
 * update its memory bookkeeping. The underlying memory is not released.
 */
public interface NativeByteSink extends QuietCloseable {
    /**
     * Get the raw pointer to the `questdb_byte_sink_t` C structure.
     */
    long ptr();
}
```

Reply:
Why do we need the pointer anywhere outside of tests? The base sink provides ways to read its contents, as well as low and high pointers.